akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -251,6 +238,36 @@ object DistributedPubSubMediator {
     @SerialVersionUID(1L)
     final case class SendToOneSubscriber(msg: Any)
 
+    /*
+     * Messages used to encode protocol to make sure that we do not send Subscribe/Unsubscribe message to 
+     * child (mediator -> topic, topic -> group) during a period of transition. Protects from situations like:
+     * 
+     * Sending Subscribe/Unsubscribe message to child actor after child has been terminated 
+     * but Terminate message did not yet arrive to parent.
+     *
+     * Sending Subscribe/Unsubscribe message to child actor that has Prune message queued and pruneDeadline set.
+     *
+     * In both of those situation parent actor still thinks that child actor is alive and forwards messages to it resulting in lost ACKs.
+     */
+    /**
+     * Passivate-like message sent from child to parent, used to signal that sender has no subscribers and no child actors.
+     */
+    case object NoMoreSubscribers
+
+    /**
+     * Sent from parent to child actor to signalize that messages are being buffered. When received by child actor
+     * if no [[Subscribe]] message has been received after sending [[NoMoreSubscribers]] message child actor will stop itself.
```

Comment:
As for the javadoc problems I tried to invoke `genjavadoc:doc` task on my local machine and the problem seems to be the placement of `PerGroupingBuffer`.

I am not sure if that is the direct reason but because `DistributedPubSubMediator` has to have `PerGroupingBufer` there is `import akka.cluster.pubsub.DistributedPubSubMediator.Internal.PerGroupingBuffer` and looking at the stacktrace 

```Scala```
error]   symbol:   class Internal
[error]   location: class DistributedPubSubMediator
[error] /home/adebski/Nauka/akka/akka-cluster-tools/target/java/akka/cluster/pubsub/DistributedPubSubMediator.java:69: error: cyclic inheritance involving DistributedPubSubMediator
[error] public  class DistributedPubSubMediator implements akka.actor.Actor, akka.actor.ActorLogging, akka.cluster.pubsub.DistributedPubSubMediator.Internal.PerGroupingBuffer {
[error]         ^
[error] /home/adebski/Nauka/akka/akka-cluster-tools/target/java/akka/cluster/pubsub/DistributedPubSubMediator.java:263: error: cannot find symbol
[error]     public static final Internal$ MODULE$ = null;
```
it may be it. 

I can just move this trait outside DistributedPubSubMediator companion object or even to separate file.